WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -768,9 +886,52 @@ public void onClick(View view) {
             mRecyclerView.setRefreshing(true);
         }
 
+        mSubFilterComponent = inflater.inflate(R.layout.subfilter_component, rootView, false);
+
+        if (BuildConfig.INFORMATION_ARCHITECTURE_AVAILABLE && mIsMainReader) {
+            mRecyclerView.getAppBarLayout().addView(mSubFilterComponent);
+
+            mSettingsButton = mSubFilterComponent.findViewById(R.id.filter_settings_button);
+            mSettingsButton.setOnClickListener(v -> {
+                showSettings();
+            });
+
+            mSubFiltersListButton = mSubFilterComponent.findViewById(R.id.filter_selection);
+            mSubFiltersListButton.setOnClickListener(v -> {
+                mBottomSheet = new SubfilterBottomSheetFragment();
+                mBottomSheet.show(getFragmentManager(), SUBFILTER_BOTTOM_SHEET_TAG);
+            });
+
+            mSubFilterTitle = mSubFilterComponent.findViewById(R.id.selected_filter_name);
+        }
+
         return rootView;
     }
 
+    private void showSettings() {
+        ReaderActivityLauncher.showReaderSubs(getActivity());
+    }
+
+    private boolean isFollowing() {
```

Comment:
Could we maybe pull this logic somewhere else? We can pass `mCurrentTag` and a nullable `currentFilter` as parameters. Maybe to ReaderUtils? Or something injected?

Reply:
done here cb0b1590629e59f73babc5f04b9522100bf4d687